src/Tests/Microsoft.NET.Build.Tests/AppHostTests.cs

```diff
@@ -21,12 +21,40 @@ namespace Microsoft.NET.Build.Tests
 {
     public class AppHostTests : SdkTest
     {
+        private static string[] GetExpectedFilesFromBuild(TestAsset testAsset)
+        {
+            var testProjectName = testAsset.TestProject?.Name ?? testAsset.Name;
+            var expectedFiles = new List<string>()
+            {
+                $"{testProjectName}{Constants.ExeSuffix}",
+                $"{testProjectName}.dll",
+                $"{testProjectName}.pdb",
+                $"{testProjectName}.deps.json",
+                $"{testProjectName}.runtimeconfig.json"
+            };
+
+            var tfm = testAsset.TargetFrameworkMoniker;
+            if (!string.IsNullOrEmpty(tfm) && (tfm.Contains("netcoreapp") || tfm.Contains("net5") || tfm.Contains("net6")))
+            {
+                var netCoreTFM = new Version(new string(tfm.Where(c => !char.IsLetter(c)).ToArray()));
```

Comment:
I think we should use NuGet TargetFramework parsing APIs (`NuGetFramework.Parse` I believe) here instead of ad-hoc manual parsing.